server/src/main/java/io/pravega/schemaregistry/service/Config.java

```diff
@@ -0,0 +1,225 @@
+import io.pravega.common.util.TypedProperties;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Properties;
+
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+/**
+ * The configuration values are retrieved using the following order, with every one overriding previously loaded values:
+ * 1. The configuration file. By default a 'schema-registry.config.properties' file is sought in the classpath; this can be
+ * overridden by setting the 'conf.file' system property to point to another one.
+ * 2. Environment Variables ({@link System#getenv()}).
+ * 3. System Properties ({@link System#getProperties()}.
+ * 4. All currently loaded values will be resolved against themselves.
+ * 5. Anything which is not supplied via the methods above will be defaulted to the values defined in this class.
+ *
+ * Configuration values can be resolved against themselves by referencing them using a special syntax. Any value
+ * of the form '${CFG}' will lookup the already loaded config value with key 'CFG' and, if defined and non-empty, it will
+ * use that config value as the final value (if not defined or empty, it will not be included in the final result and the
+ * default value (step 5) will be used). Chained resolution is not supported (i.e., CFG1=${CFG2};CFG2=${CFG3} will not
+ * set CFG1 to the value of CFG3).
+ */
+@Slf4j
+public final class Config {
+
+    //#region Properties
+
+    public static final String SERVICE_HOST;
+    public static final int SERVICE_PORT;
+    public static final int THREAD_POOL_SIZE;
+    public static final String STORE_TYPE;
+
+    public static final String PRAVEGA_CONTROLLER_URI;
+    public static final String PRAVEGA_CREDENTIALS_AUTH_METHOD;
+    public static final String PRAVEGA_CREDENTIALS_AUTH_TOKEN;
+    public static final String PRAVEGA_TLS_TRUST_STORE;
+    public static final boolean PRAVEGA_TLS_VALIDATE_HOSTNAME;
+    
+    public static final boolean TLS_ENABLED;
+    public static final String TLS_KEY_FILE;
+    public static final String TLS_KEY_PASSWORD_FILE;
+    public static final String TLS_CERT_FILE;
+
+    public static final boolean AUTH_ENABLED;
+    public static final String AUTH_RESOURCE_QUALIFIER;
+    public static final String USER_PASSWORD_FILE;
+    public static final boolean DISABLE_BASIC_AUTHENTICATION;
+
+    public static final ServiceConfig SERVICE_CONFIG;
+
+    //endregion
+
+    //region Property Definitions
+    private static final String NULL_VALUE = "{null}";
+    private static final Property<String> PROPERTY_REST_IP = Property.named("service.rest.published.host.nameOrIp", "0.0.0.0");
+    private static final Property<Integer> PROPERTY_REST_PORT = Property.named("service.rest.listener.port", 9092);
+
+    private static final Property<String> PROPERTY_STORE_TYPE = Property.named("service.store.type.name", "Pravega");
```

Comment:
* `service.store.type.name` -> `store.type.name` (as its not a configuration for the networked service)
* `service.store.pravega.controller.connect.uri` -> `store.pravega.controller.connect.uri`
* `service.store.pravega.credentials.security.auth.method` -> `store.pravega.controller.connect.security.auth.method`
* `service.store.pravega.credentials.security.auth.token` -> `store.pravega.controller.connect.security.auth.token`
* `service.store.pravega.security.tls.trustStore.location` -> `store.pravega.controller.connect.security.tls.trustStore.location`
* `service.store.pravega.security.tls.validate.hostName` -> `store.pravega.controller.connect.security.tls.validateHostName.enable`
* `service.threadPool.size` -> `threadPool.size` (as its not a property of theexposed service itself)
* `security.tls.enabled` -> `security.tls.enable` (the standard for booleans -`enable`)